internal/services/machinelearning/machine_learning_workspace_network_outbound_rule_service_tag_resource.go

```diff
@@ -0,0 +1,344 @@
+			var model machineLearningWorkspaceServiceTagOutboundRuleModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.MachineLearning.ManagedNetwork
+			id, err := managednetwork.ParseOutboundRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.SettingsRuleGet(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", id)
+			}
+
+			if existing.Model.Properties == nil {
+				return fmt.Errorf("retrieving %s: `properties` was nil", id)
+			}
+
+			payload := existing.Model
+
+			serviceTagOutboundRule := managednetwork.ServiceTagOutboundRule{
+				Type:        managednetwork.RuleTypeServiceTag,
+				Category:    pointer.To(managednetwork.RuleCategoryUserDefined),
+				Destination: &managednetwork.ServiceTagDestination{},
+			}
+
+			if metadata.ResourceData.HasChange("service_tag") {
+				serviceTagOutboundRule.Destination.ServiceTag = pointer.To(model.ServiceTag)
+			}
+
+			if metadata.ResourceData.HasChange("protocol") {
+				serviceTagOutboundRule.Destination.Protocol = pointer.To(model.Protocol)
+			}
+
+			if metadata.ResourceData.HasChange("port_ranges") {
+				serviceTagOutboundRule.Destination.PortRanges = pointer.To(model.PortRanges)
+			}
+
+			payload.Properties = serviceTagOutboundRule
+			if err := client.SettingsRuleCreateOrUpdateThenPoll(ctx, *id, *payload); err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+			return nil
+		},
+	}
+}
+
+func (r WorkspaceNetworkOutboundRuleServiceTag) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MachineLearning.ManagedNetwork
+
+			id, err := managednetwork.ParseOutboundRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.SettingsRuleGet(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			model := resp.Model
+			if model == nil {
+				return fmt.Errorf("retrieving %s: model was nil", id)
+			}
+
+			state := machineLearningWorkspaceServiceTagOutboundRuleModel{
+				Name: id.OutboundRuleName,
+			}
+
+			if props := model.Properties; props != nil {
+				if prop, ok := props.(managednetwork.ServiceTagOutboundRule); ok && prop.Destination != nil {
+					if prop.Destination.ServiceTag != nil {
+						state.ServiceTag = *prop.Destination.ServiceTag
+					}
+
+					if prop.Destination.Protocol != nil {
+						state.Protocol = *prop.Destination.Protocol
+					}
+
+					if prop.Destination.PortRanges != nil {
+						state.PortRanges = *prop.Destination.PortRanges
+					}
+				}
+			}
```

Comment:
Please review [how we nil check model and properties in the read when setting values into state in the contributor guidelines](https://github.com/hashicorp/terraform-provider-azurerm/blob/0a9f5c2a5d53ffccba36b0c1a8c184aa075f238f/contributing/topics/guide-new-resource.md?plain=1#L341-L356) and update this accordingly